source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,250 @@
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you will learn how to run a **text search** in MongoDB.
+
+You can use a text search to retrieve documents that contain a **term**
+or a **phrase** in a specified field. A term is a sequence of characters
+that excludes whitespace characters. A phrase is a sequence of terms
+that with any number of whitespace characters. 
+
+The following sections show you how to perform the following types of
+text searches:
+
+- Search Text by a Term
+- Search Text by a Phrase
+- Search Text with Terms Excluded
+
+If you want to sort by relevance, see our :ref:`Sort Results
+<sorts-crud-text-search>` page.
+
+Sample Documents
+~~~~~~~~~~~~~~~~
+
+The following sections feature examples that run text searches on the
+``fast_and_furious_movies`` collection. Each section uses a variable
+named "collection" to refer to the ``MongoCollection`` instance of the
+``fast_and_furious_movies`` collection. 
+
+The collection contains documents with a title and tags associated
+with one of the several movies that are part of the Fast and Furious
+movie franchise: 
+
+.. code-block:: json
+
+   { "_id": 1, "title": "2 Fast 2 Furious ", "tags": ["undercover", "drug dealer"] } 
+   { "_id": 2, "title": "Fast 5", "tags": ["bank robbery", "full team"] }
+   { "_id": 3, "title": "Furious 7", "tags": ["emotional"] }
+   { "_id": 4, "title": "The Fate of the Furious", "tags": ["betrayal"] }
+
+Text Index
+----------
+
+You must create a **text index** before running a text search. A text
+index specifies the string or string array field on which to run a text
+search. A collection can only contain one text index.
+
+In the following examples, you run text searches on the ``title``
+field in the ``fast_and_furious_movies`` collection. To enable text
+searches on the ``title`` field, create a text index using the
+:ref:`Indexes <index-text-indexes>` builder with the following 
+snippet: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+Multiple Indexes
+~~~~~~~~~~~~~~~~
+
+If you want to run text searches on both the ``title`` and
+``tags`` field in the ``fast_and_furious_movies`` collection, you
+need to create an text index to include them.
+
+You can create a :ref:`compound index <index-compound-indexes>`. A
+compound index is a single index that refers to multiple
+fields within a collection's documents. This allows you to specify
+multiple text fields using one text index.
+
+To enable text searches on the ``title`` and ``tag`` fields, create a
+compound index using the following snippet:  
+
+.. code-block:: java
+
+   collection.createIndex(Indexes.compoundIndex(Indexes.text("title"), Indexes.text("tags")));
+
+.. note::
+
+   If you have an existing text index on the collection, you need to drop
+   it. You can drop an index by calling the ``dropIndex()`` method,
```

Comment:
Currently sounds like this is asking to drop any existing text index regardless of what you're trying to do.

```suggestion
   Since you can only have one text index on your collection, you need to drop 
   any existing text indexes to create a new one.
   You can drop an index by calling the ``dropIndex()`` method,
```

Reply:
Where can the reader find out how to drop an index or create a compound index? I thought that could be important. If it's located somewhere else, can you link to it in this document?

Reply:
Compound text indexes have a few restrictions that I don't think are listed on that page. For more info on those restrictions:
https://docs.mongodb.com/manual/core/index-text/#std-label-text-index-compound

I think what you had in the section was good info, just needed some additional info.

Reply:
Instead of bringing over existing content, I linked to that manual page.